File: public/api/esp32/ctrl.php
<?php
// Project_Q/public/api/esp32/ctrl.php
require __DIR__ . '/../../../src/bootstrap.php';

$ip  = $_GET['ip']  ?? '';
$var = $_GET['var'] ?? '';
$val = $_GET['val'] ?? '';

if ($ip === '' || $var === '' || $val === '') {
    json_out(['ok' => false, 'error' => 'Missing ip/var/val'], 422);
}

$url = "http://{$ip}/control?var=" . urlencode($var) . "&val=" . urlencode($val);

$ch = curl_init($url);
curl_setopt_array($ch, [
    CURLOPT_RETURNTRANSFER => true,
    CURLOPT_CONNECTTIMEOUT => 3,
    CURLOPT_TIMEOUT        => 5,
]);
$body = curl_exec($ch);
$err  = curl_error($ch);
$code = curl_getinfo($ch, CURLINFO_HTTP_CODE) ?: 0;
curl_close($ch);

if ($body === false || $code < 200 || $code >= 300) {
    json_out(['ok' => false, 'error' => $err ?: "ESP32 returns HTTP {$code}"], 502);
}

json_out(['ok' => true, 'resp' => $body, 'url' => $url]);

File: public/api/esp32/ctrl.py
"""
ESP32 Control API - Python Version
Chuyển đổi từ esp32/ctrl.php
"""
import sys
import os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..', '..', '..', 'src'))

from flask import request
import requests
from helpers import json_out

def ctrl():
    """Điều khiển ESP32"""
    ip = request.args.get('ip', '')
    var = request.args.get('var', '')
    val = request.args.get('val', '')
    
    if not ip or not var or not val:
        return json_out({'ok': False, 'error': 'Missing ip/var/val'}, 422)
    
    url = f"http://{ip}/control?var={var}&val={val}"
    
    try:
        response = requests.get(url, timeout=5)
        
        if response.status_code < 200 or response.status_code >= 300:
            return json_out({'ok': False, 'error': f"ESP32 returns HTTP {response.status_code}"}, 502)
        
        return json_out({'ok': True, 'resp': response.text, 'url': url})
    except Exception as e:
        return json_out({'ok': False, 'error': str(e)}, 502)
